### ISSUE TITLE
validate.py embedded double quotes in message

### ISSUE THREAD

**Issue by cadgiru** [IfcOpenShell/IfcOpenShell#2482]
with validate.py 
In some cases message will contain double quotes
Typically something like
`"instance": "#72097=IfcFurnitureType('0ReMV0A4n08e2E7tzHAXOi',#21,'Tv og benk:46 “',$,$,(#72098,#72099,#72100),(#72095,#72096),'1352418',$,$)" `
How to, escape these
in:
- validate.py
- python

**Comment by aothms** [comment]
So, just to be clear here, you're talking about `“` which is not the double quote used to being and end json string `"`?

This is indeed somewhat unfortunate. We have in IfcOpenShell (a) the readable string formatting (what you see here, so spf escape sequences converted into unicode codepoints and entity names in schema's camel case (b) the serialized formatting, uppercase entity names and express escape sequences for non-ascii characters. Unfortunately in Python there isn't really way to trigger format (b) except for writing to a file.

So basically you're suggestion we should (have an option to) use format (b) in python and validate.py?

**Comment by aothms** [comment]
With cd27d487011a9ab91712b23817c8939b23fb566f

```
>>> import ifcopenshell
>>> f = ifcopenshell.file()
>>> f.createIfcWall(ifcopenshell.guid.new(), Name=chr(0x1F37A))
#1=IfcWall('0SvZfmyv560wI_i1uY1XRp',$,'🍺',$,$,$,$,$,$)
>>> f.createIfcWall(ifcopenshell.guid.new(), Name=chr(0x1F37A)).to_string()
"#2=IFCWALL('2VQrOGXR9FwfiTeRKBFuFU',$,'\\X2\\F37A\\X0\\',$,$,$,$,$,$)"
```

**Comment by aothms** [comment]
Edit: pasting this, I notice 0x1F37A maybe != `\X2\F37A\X0\`

**Comment by BIMvoice** [comment]
Thanks @cadgiru. On current v0.8.0 the `--json` output is produced with `json.dumps`, which escapes embedded double quotes (`"` becomes `\"`), backslashes, and newlines. Driven check: an IfcFurnitureType with Name = `Tv og benk:46 " back\slash new\nline` emits a single valid JSON line where `json.loads(line)["instance"]` round-trips to the exact instance string; the same holds for `--spf` and `--fields`. The unescaped-quote output in the original 2022 report came from an older manual formatting path that has since been replaced by `json.dumps`. Closing as resolved; please reopen if you still see unescaped quotes on a current build.
